pandora/models/pandora.py

```diff
@@ -223,11 +226,26 @@ class AdItem(PlaylistModel):
     def is_ad(self):
         return True
 
-    def register_ad(self, station_id):
-        self._api_client.register_ad(station_id, self.tracking_tokens)
+    def register_ad(self, station_id=None):
+        if not station_id:
+            station_id = self.station_id
+        if self.tracking_tokens:
+            self._api_client.register_ad(station_id, self.tracking_tokens)
+        else:
+            raise ParameterMissing('No ad tracking tokens provided for '
+                                   'registration.')
 
     def prepare_playback(self):
-        self.register_ad(self.station_id)
```

Comment:
Yes that would be more convenient for us. As per note above the API may be more flexible if we raise the exception at source, and then this serves as an example of how to deal with it?

Don't feel strongly about it though.